lib/Dialect/Arc/Transforms/LowerState.cpp

```diff
@@ -486,9 +474,11 @@ LogicalResult ModuleLowering::lowerState(StateOp stateOp) {
     nonResetBuilder = ifOp.getElseBodyBuilder();
   }
 
-  auto newStateOp = nonResetBuilder.create<StateOp>(
-      stateOp.getLoc(), stateOp.getArcAttr(), stateOp.getResultTypes(), Value{},
-      Value{}, 0, materializedOperands);
+  stateOp->dropAllReferences();
```

Comment:
Why is this method called on stateOp twice (the other time on line 421). Is there some new way that a reference can be introduced between the two?

Reply:
Good point! One should be enough.